src/firefly/js/ui/QuantityInputField.jsx

```diff
@@ -0,0 +1,251 @@
+ *
+ * If there is more than one unit option, it shows a dropdown ListBoxInputFieldView to select the unit otherwise
+ * it just shows the single unit label.
+ *
+ * @param {object} props
+ * @param {string} props.unit
+ * @param {function} props.onChange
+ * @param {Array<{label:string,value:string}>} props.unitOptions
+ */
+const QuantityUnitsPicker = ({ unit, onChange, unitOptions = [], ...props }) => (
+    <Stack direction='row' alignItems='center'>
+        <Divider orientation='vertical' />
+        {unitOptions.length > 1
+            ? <ListBoxInputFieldView
+                value={unit}
+                onChange={onChange}
+                options={unitOptions}
+                multiple={false}
+                slotProps={{
+                    input: {
+                        variant: 'plain',
+                        sx: {minHeight: 'unset'},
+                    },
+                }}
+                {...props}/>
+            : <Typography sx={{px: 1}}>{unitOptions[0]?.label || unit}</Typography>
+        }
+    </Stack>
+);
+
+/**
+ * Generic component for a Quantity (= Numeric Value + Unit) Input.
+ *
+ * It's a controlled component in which state is sent through `{value, displayValue, unit, valid, message}` props and
+ * is received back through `onChange()` prop.
+ *
+ * @param {object} props
+ * @param {number} props.min
+ * @param {number} props.max
+ * @param {object} props.sx
+ * @param {object} props.slotProps
+ * @param {object} props.inputStyle
+ * @param {string} props.orientation
+ * @param {string} props.label
+ * @param {boolean} props.showFeedback
+ * @param {string} props.placeholder
+ * @param {string} props.tooltip
+ * @param {boolean} props.connectedMarker
+ *
+ * <hr>Stateful props:
+ * @param {string} props.value quantity value in `quantityBaseUnit`
+ * @param {string} props.displayValue quantity value that is input by user
+ * @param {string} props.unit quantity unit input by user
+ * @param {boolean} props.valid whether `value` is valid (i.e. satisfies min/max bounds, nullAllowed, etc.)
+ * @param {string} props.message message to display if invalid
+ * @param {function(object, {value: string, displayValue: string, unit: string}): void} props.onChange lifts the updated
+ * value, displayValue, unit to the parent that controls the state.
+ *
+ * <hr>Quantity-specific props (required):
+ * @param {string} props.quantityName the name of the quantity (for display purposes)
+ * @param {function} props.convertQuantityValue converts numeric value between units. Signature: (valueStr, fromUnit, toUnit) => string
+ * @param {function} props.quantityAsStr formats the value (in quantityBaseUnit) to given unit. Signature: (value, unit) => string
+ * @param {string} props.quantityBaseUnit the canonical base unit key to use for `value` prop. Must be one of the `quantityUnitOptions` values.
+ * @param {Array} props.quantityUnitOptions options array for the units picker (label and value pairs)
+ */
```

Comment:
I find the naming a bit confusing. I understand it may be influenced by the base `InputField`, but it’s still not very intuitive. For example, instead of [value, quantityBaseUnit] and [unit, displayValue], I would expect [value, unit] and [displayUnit, displayValue], or even better [inputUnit, inputValue].
Also, I don’t think every prop needs a `quantity` prefix.  
That said, the documentation does help clarify things. I don’t expect any changes.  Just leaving a comment.

Reply:
Another example is `quantityAsStr`. It could follow a more standard naming pattern, such as getValueLabel. This would align with the common trio of props: value, inputValue, and getValueLabel, which represent the internal value, the value shown in the input box, and the formatted value used for display.

Reply:
> I understand it may be influenced by the base InputField, but it’s still not very intuitive.

yes this is what happened. Your naming scheme does make more sense!

> Also, I don’t think every prop needs a quantity prefix.

I think I did it so as to namespace these props under quantity-specific props bucket (as opposed to other visual or stateful props) which are also required! Earlier I was intending to use an object with custom type like QuantitySpec but then I didn't plan to nest them because it makes overriding of some of these props by consumers cumbersome.

If I take out "quantity" prefix, do you have better name suggestions for quantityName, convertQuantityValue, quantityUnitOptions too?

Reply:
I would use these.
```
quantityName 			name
convertQuantityValue 	valueConverter
quantityAsStr			valueAsLabel
quantityBaseUnit 		unit
quantityUnitOptions 	unitOptions
displayValue			inputValue
unit					inputUnit
```